doc/devel/dependencies.rst

```diff
@@ -161,3 +164,83 @@ If you go this route but need to reset and rebuild to change your settings,
 remember to clear your artifacts before re-building::
 
   git clean -xfd
+
+
+.. _development-dependencies:
+
+Additional dependencies for development
+=======================================
```

Comment:
I thought we were maintaining a conda environment for these.  Is that not the case?  It would make development a lot easier for most folks....

Reply:
I have a draft PR for this at #17096. This still needs

- check that the minimum requirements are still up-to-date. Maybe we have bumped something in between.
- a complete test in a fresh environment.
- a section in `development_setup.rst`, or rather restructuring it so that the existing docs (venv/pip) is one way and conda is an alternative. On a side note, I think that moving the dependencies in this PR actually will help with the restructuring as there is less text involved.